HRInvoiceApp/HRInvoiceApp/AddCompany.xaml.cs

```diff
@@ -19,27 +20,80 @@ public partial class AddCompany : ContentPage
 
         Department department;
         KvK kvk;
-        Province province;
+        List<Province> provinces;
         User user;
         
-        public AddCompany ()
+        public AddCompany()
 		{
-			InitializeComponent ();
+			InitializeComponent();
             SystemNavigationManager.GetForCurrentView().AppViewBackButtonVisibility = AppViewBackButtonVisibility.Visible;
             db = App.Database.GetInstance();
 
             Task.Run(async () =>
             {
-                company = await db.Table<Company>().FirstOrDefaultAsync();
+                user = await db.Table<User>().FirstOrDefaultAsync();
+                provinces = await db.Table<Province>().ToListAsync();
+                if(provinces.Count == 0)
+                {
+                    // Error handling
+                }
+
+                if(user == null)
+                {
+                    await DisplayAlert("Alert", "Voeg eerst een gebruiker toe.", "OK");
+                    var modalPage = new SettingsPage();
+                    await Navigation.PushModalAsync(modalPage);
+                    return;
+                }
                 company = new Company();
+                kvk = new KvK();
+
+                ProvincePicker.ItemsSource = provinces;
+
             });
+
         }
 
         void saveCompany(object sender, EventArgs e)
         {
+            foreach (var view in addCompanyStackLayout.Children)
+            {
+                if (view is Entry)
+                {
+                    Entry entry = (Entry)view;
+                    if (string.IsNullOrWhiteSpace(entry.Text) && entry.Placeholder.Contains("*"))
+                    {
+                        DisplayAlert("Alert", "Graag alle velden met een * invullen.", "OK");
+                        return;
+                    }
+                }
+            }
+            if (KvKNumber.Text.Count() != 8 || !int.TryParse(KvKNumber.Text, out int result))
+            {
+                DisplayAlert("Alert", "Graag een correct KvK nummer invullen.", "OK");
+                return;
+            }
+            if (ProvincePicker.SelectedItem == null)
+            {
+                DisplayAlert("Alert", "Graag een provincie invullen.", "OK");
+                return;
+            }
+            if (ZipCode.Text.Count() != 6)
+            {
+                DisplayAlert("Alert", "Graag een postcode invullen.", "OK");
+                return;
+            }            
+
             Task.Run(async () =>
             {
+                kvk.KvKNumber = int.Parse(KvKNumber.Text);
+                var kvkId = await db.InsertAsync(kvk);
```

Comment:
`InsertAsync` returned 1 of 0 (1 is true, 0 is false). als je het Id wil hebben, gewoon kvk.Id doen.